src/phoenix/server/api/input_types/PerformanceMetricInput.py

```diff
@@ -0,0 +1,106 @@
+import inspect
+from dataclasses import field, replace
+from typing import KeysView, Optional
+
+import strawberry
+from strawberry import UNSET
+
+from phoenix.core.model_schema import Column
+from phoenix.metrics.mixins import EvaluationMetric, EvaluationMetricKeywordParameters
+from phoenix.server.api.types.PerformanceMetric import PerformanceMetric
+
+
+@strawberry.input
+class MetricParametersInput:
+    pos_label_str: Optional[str] = UNSET
+    pos_label_bool: Optional[bool] = UNSET
+    pos_label_int: Optional[int] = UNSET
+    sample_weight_column_name: Optional[str] = UNSET
+
+    parameters: strawberry.Private[EvaluationMetricKeywordParameters] = field(
+        init=False,
+        repr=False,
+    )
+
+    def __post_init__(self) -> None:
+        parameters = EvaluationMetricKeywordParameters()
+
+        if (
+            isinstance(self.sample_weight_column_name, str)
+            and self.sample_weight_column_name.strip()
+        ):
+            parameters = replace(
+                parameters,
+                sample_weight=Column(self.sample_weight_column_name),
+            )
+
+        if isinstance(self.pos_label_str, str) and self.pos_label_str.strip():
+            parameters = replace(
+                parameters,
+                pos_label=self.pos_label_str,
+            )
+        elif isinstance(self.pos_label_bool, bool):
+            parameters = replace(
+                parameters,
+                pos_label=self.pos_label_bool,
+            )
+        elif isinstance(self.pos_label_int, int):
+            parameters = replace(
+                parameters,
+                pos_label=self.pos_label_int,
+            )
+
+        self.parameters = parameters
+
+    def intersect(
+        self,
+        valid_parameters: KeysView[str],
+    ) -> EvaluationMetricKeywordParameters:
+        return replace(
+            self.parameters,
+            **{k: None for k in self.parameters.keys() - valid_parameters},
+        )
+
+
+@strawberry.input
+class PerformanceMetricInput:
+    metric: PerformanceMetric
+
+    predicted_column_name: str
+    actual_column_name: str
```

Comment:
Can you not make these optional? we know this from the schema right? I get that there is there are differences between the numeric and categorical but it's gonna be too hard for the caller to figure this out, particularly the UI

Reply:
OK. Agree.